_script/test.rb

```diff
@@ -0,0 +1,40 @@
+require 'minitest/autorun'
+require 'verify_couches'
+
+class TestCouches < Minitest::Test
+    def setup
+        @couches = Couches.new
+        @country_codes = File.open("country_codes.txt").read.split("\n")
+    end 
+
+    def test_compulsory_fields
+        i = 0
+        @couches.all_front_matter.each do |front_matter|
+            assert(front_matter.has_key?("city"), "City filed not found in #{@couches.couchnames[i]}")
+            assert(front_matter.has_key?("country"), "Country field not found in #{@couches.couchnames[i]}")
+            assert(front_matter.has_key?("email"), "Email field not found in #{@couches.couchnames[i]}")
+            i+=1
+        end 
+    end
+
+    def test_country_code
+        i = 0
+        @couches.all_front_matter.each do |front_matter|
+            if front_matter.has_key?("country")
+                code = (front_matter["country"])[/\s*[A-Z][A-Z]/] #Checking for 2 character country code in CAPITALS.
+                code_line = front_matter["country"]
+                assert(@country_codes.include?(code), "Invalid country code: '#{code_line}' found in #{@couches.couchnames[i]}.\nMake sure it is a valid 2 character country code in CAPS.")
+            end
+            i+=1
+        end
+    end
+
+    def test_content
+        i = 0
+        @couches.all_content.each do |content|
+            matches = content.scan(/<[Hh]\d>/) #Checking for HTML tags that look like <h1>
```

Comment:
This is bad as well. Use nokogiri to parse html. See [this answer](http://stackoverflow.com/a/1732454/368328) on why.